File: change_username.py
from get_column import get_column
from discord import *

def change_username_func(interaction, new_username, sheet):
    number_of_users = int(sheet.acell("L1").value)

    column_c = get_column("C", sheet, "digit")

    column_d = get_column("D", sheet, "digit")

    found_user = False
    user_row = -1
    for i in range(number_of_users):
        temp_id = str(column_c[i]) + str(column_d[i])
        print(f"- temp_id: {temp_id}\nauthorID: {interaction.user.id}")
        if str(temp_id) == str(interaction.user.id):
            found_user = True
            user_row = i
            print("--------------- FOUND USER ---------------")

    print(found_user)
    if found_user:
        sheet.update_acell(f"B{user_row + 2}", new_username)
        embed = Embed(
            title="Username Change",
            description=f"Your username has been changed to \"{new_username}\" successfully.",
            color=colour.Color.green()
        )
        return embed

    else:
        embed = Embed(
            title="Not registered",
            description="You must submit your username in order to change it.",
            color=colour.Color.red()
        )
        
        return embed

File: commands/whitelisting/change_username.js
import { number_of_users } from "../../google_sheets/number_of_users.js";
import { get_cells } from "../../google_sheets/get_cells.js";
import { set_cells } from "../../google_sheets/set_cells.js";

export const change_username = async (interaction, sheet) => {
    try {
        await interaction.deferReply();

        if(interaction.options.getString("new_username") === null) {
            await interaction.followUp("One of the parameters you entered is invalid, please fix it and try again.");
            return 0;
        }

        let users = await number_of_users(sheet);
        let user_ids = (await get_cells(sheet, `Sheet1!C2:C${users+1}`)).split(",");

        let user_row = -1;

        for(let i = 0; i < parseInt(users); i++) {
            if(user_ids[i] === "#"+interaction.user.id) {
                user_row = i + 2
            }
        }
        const role = interaction.guild.roles.cache.find(r => r.name === 'Player');
        if (!role) {
            interaction.followUp("the 'Player' role was not found, could not remove it from you.")
            return;
        }
        if(user_row !== -1) {
            await set_cells(sheet, `B${user_row}`, [[interaction.options.getString("new_username")]]);
            await set_cells(sheet, `D${user_row}`, [["no"]])
            await set_cells(sheet, `E${user_row}`, [["no"]])
            const member = await interaction.guild.members.fetch(interaction.user.id);
            member.roles.remove(role);
            await interaction.followUp("Your username has been updated successfully. If you were already whitelisted, you will have be rewhitelisted.");
        }
        else {
            await interaction.followUp("You must be whitelisted in order to change your username.");
        }
    }
    catch(err) {
        interaction.channel.send(`Fatal error. \n${err}`)
        console.log(`Error: ${err}`)
    }
};
